### DISCUSSION TITLE
Document enabling `gh-pages` and `.nojekyll`

### DISCUSSION THREAD

**Discussion by kousu** [crazy-max/ghaction-github-pages#203] [Q&A] [answered]
https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#static-site-generators explains

> If you want to use a static site generator other than Jekyll, disable the Jekyll build process by creating an empty file called .nojekyll in the root of your publishing source, then follow your static site generator's instructions to build your site locally.

It would be really helpful to have this documented prominently on your readme!

And also to explain whether you need to go to https://github.com/you/repo/settings/pages to enable it or if it will Just Work once you run this Action.

**Comment by kousu** [comment]
Ah okay you do document

> jekyll 	Bool 	Allow Jekyll to build your site (default true)

But this is kind of confusingly worded. Why would you want to use this Action if you weren't trying to use a different static site generator than jekyll?

Having this enabled by default causes Github to 404 all accesses to files in folders named with leading `_`s, which is unfortunate because `sphinx` uses those heavily: `_static/`, `_images/` and `_sources` are in every build it does.

**Comment by crazy-max** [comment, marked as answer]
> But this is kind of confusingly worded. Why would you want to use this Action if you weren't trying to use a different static site generator than jekyll?

This is enabled by default because that's the default behavior of GitHub Pages when an event occurs on the gh-pages branch as you found out in https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#static-site-generators

So when the action push content on your GitHub Pages branch a GitHub bot will trigger an internal workflow that will look like this:

![image](https://user-images.githubusercontent.com/1951866/149094161-ca68059c-0a3c-431e-88ed-15b39470aeb3.png)

As you can see in https://github.com/crazy-max/ghaction-github-pages/runs/4773640592?check_suite_focus=true, this workflow will pull a jekyll image to build the website on that branch because there is no `.nojekyll` file available on this branch:

![image](https://user-images.githubusercontent.com/1951866/149094398-b1f28dfe-84d4-4d28-a515-239bce8945fe.png)

If you add a `.nojekyll` it will be a simple deployment. See https://github.com/crazy-max/crazy-max.github.io/runs/4786879422?check_suite_focus=true

![image](https://user-images.githubusercontent.com/1951866/149096618-252179b9-315b-41a6-a571-59283b6eaca6.png)
